### PR TITLE
[MIG] medical_base_history: Migrated to 10.0

### PR DESCRIPTION
**Pull request by jcdrubay**

### CODE REVIEW

File: medical_base_history/models/medical_history_entry.py [hunk 1]
```diff
@@ -46,15 +47,15 @@ class MedicalHistoryEntry(models.Model):
         help='Copy of old record for history auditing',
         readonly=True,
         store=True,
-        select=False,
+        index=False,
         compute='_compute_old_record_dict',
         inverse='_write_old_record_dict',
     )
     new_record_dict = fields.Text(
```
Comment (lasley) [inline medical_base_history/models/medical_history_entry.py:54]: We can use `fields.Serialized` here to achieve the same effect here & reduce some logic. Pretty sure I didn't know that field type existed when I built this.

File: medical_base_history/views/medical_history_entry_view.xml [hunk 2]
```diff
@@ -6,84 +6,82 @@
...
+                    <h1>
+                        <field name="associated_model_name" />
+                    </h1>
+                </header>
+                <sheet>
+                    <group name="data">
+                        <field name="display_name" />
+                        <field name="associated_record_name" />
+                        <field name="user_id" />
+                        <!-- @TODO: SMD-108: Display the old_record_dict in a template -->
```
Comment (lasley) [inline medical_base_history/views/medical_history_entry_view.xml:26]: Please move this to the roadmap

File: medical_base_history/models/medical_history_entry.py [hunk 3]
```diff
@@ -5,6 +5,7 @@
 from odoo import fields, models, api, _
 from odoo.exceptions import ValidationError
 
+# TODO: Don't use pickle, use json instead
```
Comment (lasley) [inline medical_base_history/models/medical_history_entry.py:8]: This can be removed, it's in roadmap

File: medical_base_history/README.rst [hunk 4]
```diff
@@ -24,7 +24,7 @@ Known issues / Roadmap
 ======================
 
 * Deprecate for mail.tracking.value
-
+* Stop using PICKLE, use JSON.
```
Comment (lasley) [inline medical_base_history/README.rst:27]: We can remove this if we switch the fields to Serialized type.

**Review by lasley — COMMENTED** [review]
Thanks @jcdrubay - few comments

**Comment by jcdrubay** [comment]
@lasley , what do you think if we drop this module in favor of https://github.com/OCA/server-tools/tree/10.0/auditlog ?

Because it seems both this modules of the same purpose.

I think we should focus on auditlog and make it work nicely so we can get better support in future. The issue is that none of these 2 modules are available in v10.

Also, I realize now that I have other issues with this module, for instance I get this error just when clicking on the menu History > Entries ?  ` "AttributeError: 'medical.appointment' object has no attribute 'get_name'".` . I admit that I did a "quick and dirty" migration to v10 just to get medical_appointment installable...

Also, if you agree, we could temporarily drop the dependency to this module in medical_appointment.

**Comment by lasley** [comment]
@jcdrubay - I was planning to deprecate for just `mail.tracking.values`, but `auditlog` will work too. Pretty sure that's in the ReadMe RoadMap, although it might just be the tracking.values.

In terms of dropping the depends, we could do that too. medical_appointment hasn't been released on v9 yet and I'm pretty sure v8 didn't have any sort of auditing. We wouldn't technically be losing a feature that already existed, so we wouldn't be breaking any rules.

File: medical_base_history/__manifest__.py [hunk 5]
```diff
@@ -24,6 +24,6 @@
     ],
     'website': 'https://laslabs.com',
     'licence': 'AGPL-3',
```
Comment (tedsalmon) [inline medical_base_history/__manifest__.py:26]: s/licence/license

**Review by tedsalmon — APPROVED** [review]
LGTM minus a minor, pre-existing spelling issue :)

**Comment by lasley** [comment]
@jcdrubay - We ended up refactoring around this module using auditlog as was suggested. New PR is #171

**Pull request closed without merging**